### PR TITLE
L-T pairing removed (changes from G.K.) + fixes in TDCChannel class

### PR DESCRIPTION
**Pull request by alekgajos**
This PR includes a simplification of the Unpacker done by Greg and a cleanup of unused code.
Moreover, fixes of the following Redmine issues are included: 870, 905, 953.

### CODE REVIEW

File: Unpacker2/Unpacker2.cc [hunk 1]
```diff
@@ -2,29 +2,53 @@
...
 #include <map>
+#include "TimeDateDecoder.h"
 #include "Unpacker2.h"
-#include "Unpacker_TRB3.h"
-#include "Unpacker_Lattice_TDC.h"
 #include <boost/property_tree/ptree.hpp>
 #include <boost/property_tree/xml_parser.hpp>
+#include <TObjString.h>
+
+#define MAX_ALLOWED_REPETITIONS 1
```
Comment (wkrzemien) [inline Unpacker2/Unpacker2.cc]: to const static member of the class?
e.g const static int kMaxAllowedRepetitions =1;

File: Unpacker2/Unpacker2.cc [hunk 2]
```diff
@@ -2,29 +2,53 @@
...
+  
+  return s;  
 }
 
-Unpacker2::Unpacker2(const char* hldFile, const char* configFile, int numberOfEvents) {
-
-  Init();
-
-  eventsToAnalyze = numberOfEvents;
+UInt_t StringToUInt(std::string hex_string){
```
Comment (wkrzemien) [inline Unpacker2/Unpacker2.cc]: StringToUInt(const std::string& hex_string)

This maybe could be replaced by the std functions
https://en.cppreference.com/w/cpp/string/basic_string/stoul

File: Unpacker2/Unpacker2.h [hunk 3]
```diff
@@ -35,34 +35,31 @@ class Unpacker2 : public TObject {
...
   TH1F * loadCalibHisto(const char* calibFile);
+  bool loadTDCcalibFile(const char* calibFile);
   
   void ParseConfigFile(std::string f, std::string s);
-  void DistributeEvents(std::string f);
   void DistributeEventsSingleStep(std::string file);
   
-  void AddUnpacker(std::string s, UnpackingModule* u) { unpackers[s] = u; }
-  UnpackingModule* GetUnpacker(std::string s) { return unpackers[s]; }
+  //  void AddUnpacker(std::string s, UnpackingModule* u) { unpackers[s] = u; }
```
[the remark this answers is not in the document]
Comment (alekgajos) [reply Unpacker2/Unpacker2.h]: done

File: Unpacker2/Unpacker2.h [hunk 4]
```diff
@@ -35,34 +35,31 @@ class Unpacker2 : public TObject {
   // stuff for "calculate_times"
   TClonesArray* pArray = 0;
   TIter * iter;
-  TDCHit* pHit = 0;
   int refTimeEpoch[REF_CHANNELS_NUMBER];
   int refTimeCoarse[REF_CHANNELS_NUMBER];
   int refTimeFine[REF_CHANNELS_NUMBER];
   int refChannelOffset;
-  TH1F * calibHist;
+  TH1F * TOTcalibHist;
```
[the remark this answers is not in the document]
Comment (alekgajos) [reply Unpacker2/Unpacker2.h]: done

File: Unpacker2/Unpacker2.h [hunk 5]
```diff
@@ -18,10 +16,12 @@ class Unpacker2 : public TObject {
   
 private:
   
-  std::map<std::string, UnpackingModule*> unpackers;
+  std::map<UInt_t, UInt_t> tdc_offsets;
   
   int eventsToAnalyze;
   
+  bool useTDCcorrection;
```
[the remark this answers is not in the document]
Comment (alekgajos) [reply Unpacker2/Unpacker2.h]: done

File: Unpacker2/Unpacker2.cc (hunk 1 above)
Comment (alekgajos) [reply Unpacker2/Unpacker2.cc]: done

File: Unpacker2/Unpacker2.cc (hunk 2 above)
Comment (alekgajos) [reply Unpacker2/Unpacker2.cc]: You are right, std::stoul works here. Replaced.

**Comment by alekgajos** [comment]
This is the final version, ready for review.

**Pull request merged**